enumerables.rb

```diff
@@ -0,0 +1,165 @@
+    end
+  end
+
+  def my_any?(arg = nil)
+    to_change = false
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = true unless obj == false || obj.nil? }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = true if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = true if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = true if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = true if yield obj }
+    end
+    to_change
+  end
+
+  def my_all?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false unless arr.include?(true) || obj != false }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false unless obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false unless obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false unless obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false unless yield obj }
+    end
+    to_change
+  end
+
+  def my_none?(arg = nil)
+    to_change = true
+    arr = self
+    if !block_given?
+      if arg.nil?
+        my_each { |obj| to_change = false if arr.include?(true) || obj == true }
+      elsif arg.respond_to?(:to_i)
+        my_each { |obj| to_change = false if obj == arg }
+      elsif arg.is_a?(Regexp)
+        my_each { |obj| to_change = false if obj.match(arg) }
+      elsif arg.respond_to?(:class)
+        my_each { |obj| to_change = false if obj.instance_of? arg }
+      end
+    else
+      my_each { |obj| to_change = false if yield obj }
+    end
+    to_change
+  end
+
+  def my_map
+    arr = self
+    return arr.to_enum unless block_given?
+
+    to_change = []
+    my_each do |obj|
+      to_change.push(yield obj)
+    end
+    to_change
+  end
+
+  def my_count(arg = nil)
+    arr = self
+    total = 0
+    if arg.nil?
+      if block_given?
+        my_each do |item|
+          total += 1 if yield(item)
+        end
+      else
+        total = arr.length
+      end
+    else
+      my_each { |obj| total += 1 if obj == arg }
+    end
+    total
+  end
+
+  def my_inject(start_value = 0, operator = nil)
+    arr = self
+    to_change = 0
+    to_change = 1 if operator == :*
+    to_change = 1 if operator == :/
+    to_change = start_value if start_value != 0
+    operator = start_value if start_value.is_a? Symbol
+    if operator.is_a? Symbol
+      case operator
+      when :+
+        arr.each { |obj| to_change += obj }
+      when :-
+        arr.each { |obj| to_change -= obj }
+      when :*
+        arr.each { |obj| to_change *= obj }
+      when :/
+        arr.each { |obj| to_change /= obj }
+      end
+    end
+    if block_given?
+      my_each do |obj|
+        to_change = yield to_change, obj
+      end
+    end
+    to_change
+  end
+end
```

Comment:
`my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
%w[dog door rod blade].my_inject {|memo, word| memo.length > word.length ? memo : word}  #should return 'blade'
```
Check your to_check for this criteria. The answer is in there.
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(:+)  #should return 6
```
But while I tried to execute, I got an error.